olps/visualization/visualizer.py

```diff
@@ -1,30 +1,15 @@
 import matplotlib.pyplot as plt
 from olps.strategies.strategy import Strategy
-from typing import *
+from abc import ABC, abstractmethod
 
-class Visualizer:
+class Visualizer(ABC):
     """
-    Class for visualizing cumulative wealth of multiple strategies
+    Abstract class for visualizing portfolios over time
     """
-    outputfile: str
-    strategies: list[Strategy]
-
-    def __init__(self, outputfile: str, strategies: list[Strategy]):
-        """
-        Initialize the visualizer with already run strategies and output file
-        """
-        self.outputfile = outputfile
-        self.strategies = strategies
```

Comment:
If you're creating a constructor to store the output file and strategy list, how come it's not used anywhere? `visualize()` is a static method and won't make use of instance variables

Reply:
The constructor was deleted, so these are removed lines